test/eventmesh.LockManager.spec.js

```diff
@@ -0,0 +1,176 @@
+        const noLockResp = {
+          'count': 0,
+          'operationType': ''
+        };
+        jsonStub.onCall(0).returns(noLockResp);
+        return manager.isWriteLocked('fakeResource')
+          .then(result => {
+            /* jshint expr: true */
+            expect(result).to.eql(false);
+            expect(getStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+          });
+      });
+    });
+
+    describe('#lock', () => {
+      it('should succeed when lock details is undefined.', () => {
+        const writeLockResp = {
+          'count': 1,
+          'operationType': 'WRITE'
+        };
+        return manager.lock('fakeResource', CONST.LOCK_TYPE.WRITE)
+          .then(() => {
+            /* jshint expr: true */
+            expect(lockStub.getCall(0).calledWithExactly('fakeResource/lock')).to.be.true;
+            expect(putStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+            expect(valueStub.getCall(0).calledWithExactly(JSON.stringify(writeLockResp))).to.be.true;
+            expect(getStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+            expect(acquireStub.called).to.be.true;
+            expect(releaseStub.called).to.be.true;
+          });
+      });
+      it('should succeed if no ongoing lock is there.', () => {
+        const noLockResp = {
+          'count': 0,
+          'operationType': ''
+        };
+        const writeLockResp = {
+          'count': 1,
+          'operationType': 'WRITE'
+        };
+        jsonStub.onCall(0).returns(noLockResp);
+        return manager.lock('fakeResource', CONST.LOCK_TYPE.WRITE)
+          .then(() => {
+            /* jshint expr: true */
+            expect(lockStub.getCall(0).calledWithExactly('fakeResource/lock')).to.be.true;
+            expect(putStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+            expect(valueStub.getCall(0).calledWithExactly(JSON.stringify(writeLockResp))).to.be.true;
+            expect(getStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+            expect(acquireStub.called).to.be.true;
+            expect(releaseStub.called).to.be.true;
+          });
+      });
+      it('should fail if an ongoing lock is there.', () => {
+        const writeLockResp = {
+          'count': 1,
+          'operationType': 'WRITE'
+        };
+        jsonStub.onCall(0).returns(writeLockResp);
+        return manager.lock('fakeResource', CONST.LOCK_TYPE.WRITE)
+          .catch(e => {
+            /* jshint expr: true */
+            expect(e.message).to.eql('Could not acquire lock for fakeResource');
+            expect(lockStub.getCall(0).calledWithExactly('fakeResource/lock')).to.be.true;
+            expect(getStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+            expect(acquireStub.called).to.be.true;
+            expect(releaseStub.called).to.be.true;
```

Comment:
Same comment: Since there are 2 places where lock.release() is called, can you check that it is called exactly once?

Reply:
Done.